### ISSUE TITLE
Is it possible to have a websocket server at a particular route?

### ISSUE THREAD

**Issue by habemus-papadum** [JuliaWeb/WebSockets.jl#22]
Hi -- Based on the examples, it wasn't clear to me how create a websocket server which was attached to a particular "route" e.g. on the client side I would like the code to look like:
`var ws = new WebSocket("ws://localhost:9998/echo");` 

rather than 

`var ws = new WebSocket("ws://localhost:9998");`

and ideally I could have multiple websocket servers listening on different routes.

Does anyone know if this easily possible given the current architecture?

thanks!

**Comment by hustf** [comment]
Nabemus, what you're looking for is subprotocols (which go under various names).

Currently you may want to Pkg.checkout("WebSockets.jl") since we're a little behind on tagging versions.

In Julia, before you define the server / web socket handlers:
```
	WebSockets.addsubproto("echo")
```

In your Julia websockethandler(wsrequest, websocket):
```
function websockethandler(wsrequest, websocket)
		if 	wsrequest.headers["Sec-WebSocket-Protocol"] == "echo"
			ws_echo(websocket)
                else
                       ws(websocket)
                end
end
```


In javascript:
```
        var wsuri = document.URL.replace("http:","ws:");
	return new WebSocket(wsuri, "echol")
```

**Comment by habemus-papadum** [comment]
thanks! is the trailing `l` in the javascript `"echol"` a typo?

**Comment by hustf** [comment]
It is.
See the current pull request for a full example.

**Comment by hustf** [comment]
Closed by v0.3.0. We need tests for binary sockets, so why not contribute with a websocket for text and another specialize on binary data / mouse interactions?
